code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageFragment.java

```diff
@@ -91,10 +91,14 @@ public boolean onTouch(final View view, final MotionEvent motionEvent) {
 
         // determine if the tapped view is the webview
         if (view.getId() == message.webView.getId()) {
-            // pass the event to the gesture detector to determine if a motion event occurred on the
-            // webview.
-            gestureDetector.onTouchEvent(motionEvent);
-            // perform the tap to allow interaction with buttons within the webview
+            // if we have no gestures just pass the touch event to the webview
+            if (!message.getMessageSettings().getGestures().isEmpty()) {
```

Comment:
two things here.  

- do we know that gestures returned from `getGestures` is never going to be null?  might be good to do a null check.  
- ideally the code matches the comments (comment says if no gestures then pass through, but the condition checks for the presence of gestures then implements that case).  here it's probably easier to read if you have the early exit (no gestures) at the top of the method, then the rest of the method doesn't need to be wrapped in the `if` block.